libi2pd_client/HTTPProxy.cpp

```diff
@@ -219,7 +219,9 @@ namespace proxy {
 		/* replace headers */
 		req.UpdateHeader("User-Agent", "MYOB/6.66 (AN/ON)");
 		/* add headers */
-		req.UpdateHeader("Connection", "close"); /* keep-alive conns not supported yet */
+		auto h = req.GetHeader ("Connection");
+		if (h.find("upgrade") == std::string::npos && h.find("Upgrade") == std::string::npos)
```

Comment:
Better to try to find "pgrade" and then compare lowercase of [-1] with 'u'

Reply:
Something like that?

`if (std::tolower(h[x - 1], std::locale("C")) == 117) do_stuff();`

Reply:
check now